iommi/part.py

```diff
@@ -157,7 +157,9 @@ def dispatch_response_handler(r):
             if request.method == 'POST':
                 assert False, 'This request was a POST, but there was no dispatch command present.'
 
-        return HttpResponse(render_root(part=self, **kwargs))
+        response = HttpResponse(render_root(part=self, **kwargs))
+        response.iommi_part = self
```

Comment:
Nice! I could have used this while writing tests for other stuff today. We had tests taking advantage of some code piggy-backing the context on the response in the same way, but getting the actual iommi thing would be even better :)